llm/cli.py

```diff
@@ -163,33 +156,55 @@ def prompt(
             prompt, system = template_obj.execute(prompt, params)
         except Template.MissingVariables as ex:
             raise click.ClickException(str(ex))
-        if model is None and template_obj.model:
-            model = template_obj.model
-
-    messages = []
-    if _continue:
-        _continue = -1
-        if chat_id:
-            raise click.ClickException("Cannot use --continue and --chat together")
-    else:
-        _continue = chat_id
-    chat_id, history = get_history(_continue)
+        if model_id is None and template_obj.model:
+            model_id = template_obj.model
+
     history_model = None
-    if history:
-        for entry in history:
-            if entry.get("system"):
-                messages.append({"role": "system", "content": entry["system"]})
-            messages.append({"role": "user", "content": entry["prompt"]})
-            messages.append({"role": "assistant", "content": entry["response"]})
-            history_model = entry["model"]
-    if system:
-        messages.append({"role": "system", "content": system})
-    messages.append({"role": "user", "content": prompt})
-    if model is None:
-        model = history_model or DEFAULT_MODEL
+    # TODO: Re-introduce --continue mode
+    # messages = []
+    # if _continue:
+    #     _continue = -1
+    #     if chat_id:
+    #         raise click.ClickException("Cannot use --continue and --chat together")
+    # else:
+    #     _continue = chat_id
+    # chat_id, history = get_history(_continue)
+    # history_model = None
+    # if history:
+    #     for entry in history:
+    #         if entry.get("system"):
+    #             messages.append({"role": "system", "content": entry["system"]})
+    #         messages.append({"role": "user", "content": entry["prompt"]})
+    #         messages.append({"role": "assistant", "content": entry["response"]})
+    #         history_model = entry["model"]
+
+    # Figure out which model we are using
+    if model_id is None:
+        model_id = history_model or DEFAULT_MODEL
+
+    # Now resolve the model
+    try:
+        model = model_aliases[model_id]
+    except KeyError:
+        raise click.ClickException("'{}' is not a known model".format(model_id))
+
+    # TODO: Only do this for OpenAI models
+    openai.api_key = get_key(key, "openai", "OPENAI_API_KEY")
```

Comment:
If I do that though it conflicts a bit with how the CLI tool works, since that gets a dict of model instances back from `get_model_aliases()`. These have already been instantiated, so when do they get passed their key?

Reply:
I like the idea that a model subclass (`Chat` in this case) knows if it needs a key AND knows what the default name of that key (the name stored in the `keys.json` file) should be.

That way you can either pass it a key when you instantiate it, or there's a mechanism by which the CLI harness knows how to give it a key later.